container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/logs_aggregator_functions/package_private_helpers.go

```diff
@@ -0,0 +1,105 @@
+	}
+
+	logsAggregatorContainer := allLogsAggregatorContainers[0]
+	logsAggregatorContainerID := logsAggregatorContainer.GetId()
+
+	logsAggregatorObject, err := getLogsAggregatorObjectFromContainerInfo(
+		ctx,
+		logsAggregatorContainerID,
+		logsAggregatorContainer.GetStatus(),
+		dockerManager,
+	)
+	if err != nil {
+		return nil, "", stacktrace.Propagate(err, "An error occurred getting the logs Aggregator object using container ID '%v', labels '%+v' and the status '%v'", logsAggregatorContainer.GetId(), logsAggregatorContainer.GetLabels(), logsAggregatorContainer.GetStatus())
+	}
+
+	return logsAggregatorObject, logsAggregatorContainerID, nil
+}
+
+func getAllLogsAggregatorContainers(ctx context.Context, dockerManager *docker_manager.DockerManager) ([]*types.Container, error) {
+	logsAggregatorContainerSearchLabels := map[string]string{
+		label_key_consts.AppIDDockerLabelKey.GetString():         label_value_consts.AppIDDockerLabelValue.GetString(),
+		label_key_consts.ContainerTypeDockerLabelKey.GetString(): label_value_consts.LogsAggregatorTypeDockerLabelValue.GetString(),
+	}
+
+	matchingLogsAggregatorContainers, err := dockerManager.GetContainersByLabels(ctx, logsAggregatorContainerSearchLabels, shouldShowStoppedLogsAggregatorContainers)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred fetching logs Aggregator containers using labels: %+v", logsAggregatorContainerSearchLabels)
+	}
+	return matchingLogsAggregatorContainers, nil
+}
+
+func getLogsAggregatorObjectFromContainerInfo(
+	ctx context.Context,
+	containerId string,
+	containerStatus types.ContainerStatus,
+	dockerManager *docker_manager.DockerManager,
+) (*logs_aggregator.LogsAggregator, error) {
+
+	privateIpAddr := net.IP{}
+
+	isContainerRunning, found := consts.IsContainerRunningDeterminer[containerStatus]
+	if !found {
+		// This should never happen because we enforce completeness in a unit test
+		return nil, stacktrace.NewError("No is-running designation found for logs aggregator container status '%v'; this is a bug in Kurtosis!", containerStatus.String())
+	}
+	var logsAggregatorStatus container_status.ContainerStatus
+	if isContainerRunning {
+		logsAggregatorStatus = container_status.ContainerStatus_Running
+
+		privateIpAddrStr, err := dockerManager.GetContainerIP(ctx, consts.NameOfNetworkToStartEngineAndLogServiceContainersIn, containerId)
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "An error occurred getting the private IP address of container '%v' in network '%v'", containerId, consts.NameOfNetworkToStartEngineAndLogServiceContainersIn)
+		}
+		privateIpAddr = net.ParseIP(privateIpAddrStr)
+		if privateIpAddr == nil {
+			return nil, stacktrace.NewError("Couldn't parse private IP address string '%v' to an IP", privateIpAddrStr)
+		}
+	} else {
+		logsAggregatorStatus = container_status.ContainerStatus_Stopped
+	}
+
+	logsAggregatorObj := logs_aggregator.NewLogsAggregator(
+		logsAggregatorStatus,
+		privateIpAddr,
```

Comment:
I think here if the container is stopped the IpAddress field will contain a value (an empty IP address). Should we instead set it to nil? You just need to declare your IP address with `var privateIpAddr net.IP` at the top of this function

Reply:
yep agreed, good catch